### PR TITLE
Move `Cargo.toml` templating to `Cargo.toml.template`

### LINKED ISSUE #10: Liquid placeholders make `cargo` and `rust-analyzer` fail
For example, `"{{project-name}}"` in `Cargo.toml` is not a valid crate name, so the template itself cannot be compiled. This makes it harder for maintainers of the template to make changes.

Generated repos are not affected by this issue, only the template itself.

### PR DESCRIPTION
**Pull request by benfrankel**
Fixes https://github.com/TheBevyFlock/bevy_new/issues/10.

### CODE REVIEW

File: Cargo.toml.template [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Enable a small amount of optimization in the dev profile.
+[profile.dev]
+opt-level = 1
+
+# Enable a large amount of optimization in the dev profile for dependencies.
+[profile.dev.package."*"]
+opt-level = 3
+
+# Remove expensive debug assertions due to <https://github.com/bevyengine/bevy/issues/14291>
+[profile.dev.package.wgpu-types]
```
Comment (janhohenheim) [inline Cargo.toml.template:27]: Why is this included, but not the log and tracing stuff? Imo we should remove this for the minimal version.

File: cargo-generate.toml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 [template]
 cargo_generate_version = "0.22"
+include = ["*.template"]
```
Comment (janhohenheim) [inline cargo-generate.toml:3]: Why did you end up calling these template instead of liquid? Shouldn't the latter have better syntax highlighting?

File: post-generate.rhai [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Rename template files.
+file::rename("Cargo.toml.template", "Cargo.toml");
```
Comment (janhohenheim) [inline post-generate.rhai:2]: If this happens *after* the generation, are the placeholders still replaced?

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Thanks! Curious about some decisions :)

File: cargo-generate.toml (hunk 2 above)
Comment (benfrankel) [reply cargo-generate.toml:3]: If you call it `xyz.liquid`, `cargo-generate` will recognize it as a template for `xyz` and see that `xyz` already exists, so it will delete `xyz.liquid` instead. Well I don't know what `cargo-generate` is actually doing but that's the behavior I see.

File: post-generate.rhai (hunk 3 above)
Comment (benfrankel) [reply post-generate.rhai:2]: Yes, `cargo-generate` will first replace the placeholders in `Cargo.toml.template`, and then renaming the file here will overwrite the template's `Cargo.toml` with the user's expanded `Cargo.toml.template`.

File: Cargo.toml.template (hunk 1 above)
Comment (benfrankel) [reply Cargo.toml.template:27]: Well, that would be a separate PR since this is just a `cp` of `Cargo.toml`.

**Review by janhohenheim — APPROVED** [review]
